python/pyspark/context.py

```diff
@@ -493,6 +493,10 @@ def getStart(split):
                 return start0 + int((split * size / numSlices)) * step
 
             def f(split, iterator):
+                # it's an empty iterator here but we need this line for triggering the logic of
+                # checking END_OF_DATA_SECTION during load iterator in runtime, thus make sure
```

Comment:
Can you elaborate `during load iterator in runtime` please? For instance,

`FramedSerializer.load_stream` has its signal handling, for instance, `SpecialLengths.END_OF_DATA_SECTION` in `_read_with_length`. Since `FramedSerializer.load_stream` produces a generator, the control should at least be in that function once. Here we do it by explicitly converting the empty iterator to a list.